packages/contracts/src/mocks/DAOMock.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity ^0.8.8;
+
+import {IDAO} from "@aragon/osx-commons-contracts/src/dao/IDAO.sol";
+import {IPermissionCondition} from "@aragon/osx-commons-contracts/src/permission/condition/IPermissionCondition.sol";
+import {PermissionLib} from "@aragon/osx-commons-contracts/src/permission/PermissionLib.sol";
+
+contract DAOMock is IDAO {
+    // TODO Move constants into `PermissionLib`
+    address internal constant NO_CONDITION = address(0);
+
+    // TODO create `IPermissionManager` interface
+    event Granted(
+        bytes32 indexed permissionId,
+        address indexed here,
+        address where,
+        address indexed who,
+        address condition
+    );
+
+    event Revoked(
+        bytes32 indexed permissionId,
+        address indexed here,
+        address where,
+        address indexed who
+    );
+
+    bool public hasPermissionReturnValueMock;
+
+    function setHasPermissionReturnValueMock(bool _hasPermissionReturnValueMock) external {
+        hasPermissionReturnValueMock = _hasPermissionReturnValueMock;
+    }
+
+    function hasPermission(
+        address _where,
+        address _who,
+        bytes32 _permissionId,
+        bytes memory _data
+    ) external view override returns (bool) {
+        (_where, _who, _permissionId, _data);
+        return hasPermissionReturnValueMock;
+    }
+
+    function applyMultiTargetPermissions(
+        PermissionLib.MultiTargetPermission[] calldata _items
+    ) external {
+        for (uint256 i; i < _items.length; ) {
+            PermissionLib.MultiTargetPermission memory item = _items[i];
+
+            if (item.operation == PermissionLib.Operation.Grant) {
+                grant({_where: item.where, _who: item.who, _permissionId: item.permissionId});
+            } else if (item.operation == PermissionLib.Operation.Revoke) {
+                revoke({_where: item.where, _who: item.who, _permissionId: item.permissionId});
+            } else if (item.operation == PermissionLib.Operation.GrantWithCondition) {
+                grantWithCondition({
+                    _where: item.where,
+                    _who: item.who,
+                    _permissionId: item.permissionId,
+                    _condition: IPermissionCondition(item.condition)
+                });
+            }
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function grant(address _where, address _who, bytes32 _permissionId) public {
+        (_where, _who, _permissionId);
+
+        emit Granted({
+            permissionId: _permissionId,
+            here: msg.sender,
+            where: _where,
+            who: _who,
+            condition: NO_CONDITION
+        });
+    }
+
+    function revoke(address _where, address _who, bytes32 _permissionId) public {
+        (_where, _who, _permissionId);
```

Comment:
spoke about this convention with Sarkawt and Michael. Anonymous arguments avoid the need to invoke variables at the top of the code (I assume this has a minor gas cost as well), and create less noise. Not a crisis thought, especially in a mock

Reply:
The compiler is smart enough to optimize this so it doesn't cause more gass. I prefer this variant because we have explicit names for the input variables, although not used. It also has the benefit that the documentation shows these variable names (where we should point out why these are not used).
In this mock, it is not necessary, ofc.

Without `(a,b,c);`:
![Bildschirmfoto 2024-02-06 um 12 51 03](https://github.com/aragon/osx-plugin-template-hardhat/assets/20623991/5cf3b367-4d89-4700-b6c7-b17c8954c6a1)

With `(a,b,c);`:
![Bildschirmfoto 2024-02-06 um 12 51 03](https://github.com/aragon/osx-plugin-template-hardhat/assets/20623991/bcef192e-8e7e-442b-a964-61d3d0b1b090)

Reply:
makes sense thx